File: inventario_back/src/main/java/upn/pe/inventario_back/models/Salida.java
package upn.pe.inventario_back.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "salida")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Salida {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String documento;
    private String condicion;
    private String tipoPago;
    private String formapago;

    private Double total;
    private Double gravada;
    private Double impuesto;

    private LocalDateTime fecha_emision;
    private LocalDateTime fecha_vencimiento;
    private String nota;
    @ManyToOne
    @JoinColumn(name = "usuario_empresa_id", nullable = false)
    private UsuarioModel usuarioEmpresa;

    // Usuario proveedor asociado al ingreso
    @ManyToOne
    @JoinColumn(name = "usuario_cliente_id", nullable = false)
    private UsuarioModel usuarioCliente;

    @OneToMany(mappedBy = "salida", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DetalleSalida> detalles = new ArrayList<>();
}

File: inventario_front/src/app/Models/Salida.ts
export interface SalidaRequest {
  documento: string;
  condicion: string;
  tipopago: string;
  formapago: string;
  total: number;
  gravada: number;
  impuesto: number;
  fecha_emision: string;
  fecha_vencimiento: string;
  usuarioEmpresa_id: number;
  usuarioCliente_id: number;
  nota: string;
  detalles: DetalleSalidaRequest[];
}

export interface DetalleSalidaRequest {
  nombre: string;
  cantidad: number;
  precioUnitario: number;
  total: number;
  producto_id: string;
}
